packages/cli/src/util/events.ts

```diff
@@ -59,113 +61,128 @@ async function printEvents(
       if (findOpts.since) query.set('since', String(findOpts.since));
       if (findOpts.until) query.set('until', String(findOpts.until));
 
-      const eventsUrl = `/v1/now/deployments/${deploymentIdOrURL}/events?${query}`;
-      const eventsRes = await client.fetch(eventsUrl, { json: false });
-
-      if (eventsRes.ok) {
-        const readable = eventsRes.body;
-
-        // handle the event stream and make the promise get rejected
-        // if errors occur so we can retry
-        return new Promise<void>((resolve, reject) => {
-          const stream = readable.pipe(jsonlines.parse());
-
-          let poller: ReturnType<typeof setTimeout>;
-
-          if (mode === 'deploy') {
-            poller = (function startPoller() {
-              return setTimeout(async () => {
-                try {
-                  const json = await getDeployment(
-                    client,
-                    contextName,
-                    deploymentIdOrURL
-                  );
-                  if (json.readyState === 'READY') {
+      const eventsUrl = `/v3/now/deployments/${deploymentIdOrURL}/events?${query}`;
+      try {
+        const eventsRes = await client.fetch(eventsUrl, {
+          json: false,
+          // @ts-expect-error: typescipt is getting confused with the signal types from node (web & server) and node-fetch (server only)
+          signal: abortController?.signal,
+        });
+
+        if (eventsRes.ok) {
+          const readable = eventsRes.body;
+
+          // handle the event stream and make the promise get rejected
+          // if errors occur so we can retry
+          return new Promise<void>((resolve, reject) => {
```

Comment:
I took a look at this promise, and it looks mostly fine, but it's more difficult to read the code and make sure it works right in case of a retry 🤔

Reply:
If it were me, I'll completely rebuild this whole function with effects to get more readable code.

But here again, I meant to inject my changes in the existing code without compromising its shape.